### ISSUE TITLE
future work

### ISSUE THREAD

**Issue by borawhocodess** [automl/TFM-Playground#44]
running list of things:
- [ ] plug tabarena official eval as dependency ! (+beyond) [tabarena[benchmark]](https://github.com/autogluon/tabarena/blob/d6ef8474e757ef3577b16eb96bec76e57516042c/README.md?plain=1#L117)
- [ ] talent!
- [ ] turn epochs into steps 
- [ ] fold train into pretrain
- [ ] own prior eval !!!!!!!!!!!!!
- [ ] type checker !!! [ty](https://docs.astral.sh/ty/)
- [ ] items on the end of this pr https://github.com/borawhocodess/TFM-Playground-pr/pull/11 
- [ ] connect live ticl prior
- [ ] connect current tabpfnv1 prior
- [ ] priordumpdataloader pointer
- [ ] default device check/delete? 
- [ ] muon !!
- [ ] train full size tabicl (cluster) (multi seed)
- [ ] curriculum learning ? 
- [ ] agents.md
- [ ] subfolder internal external priors/models
- [ ] add [exaone](https://github.com/LGAI-Research/EXAONE-Tabular) ? 
- [ ] nanotabpfnv3 (v4 or v3.5 coming this quarter!!!)
- [ ] bring official tabpfn in for training ?
- [ ] bring official tabdpt in for training ?
- [ ] check if tabdpt prior exist
- [ ] omar/jaris [hf trainer](https://huggingface.co/docs/transformers/en/main_classes/trainer) david [torchtitan](https://github.com/pytorch/torchtitan/tree/main)
- [ ] adaptive early stopping 
- [ ] make moddednanoprior more nano (no reattempts) and bring it in
- [ ] bring our own scm prior in (git history)
- [ ] experiment: noise vs no noise training
- [ ] experiment: distance aware bucket loss 
- [ ] interpretibility of predictions [x](https://x.com/prior_labs/status/2095516427902210137) [cookbook](https://docs.priorlabs.ai/cookbook/decoder_readout)
- [ ] checkpoint resuming
- [ ] finetuning
- [ ] [tabiclv2icmltalk](https://icml.cc/virtual/2026/80470) min 18.25 tfmp and min 25.25 discussion
- [ ] check [dopfn](https://github.com/jr2021/Do-PFN) prior
- [ ] check [tempopfn](https://github.com/automl/TempoPFN)
- [ ] [is one layer enough repo](https://github.com/amirbalef/is_one_layer_enough) kind experiments
